### ISSUE TITLE
Reproducing results on BAIR

### ISSUE THREAD

**Issue by kaanakan** [edouardelasalles/srvp#7]
Hi,

I do not want to mix up the issues, I open another issue for this. Again, thank you for providing the code.

The metrics for BAIR dataset is reported (Table 5) as 19.59 ± 0.27 (PSNR), 0.8196 ± 0.0084 (SSIM). However, when I run the test code with the instructions given in the README file (I add device 0 and batch size 32) and the pre-trained weights you shared, the results from the script are:

psnr 18.830994 +/- 0.274060565829277
ssim 0.80773425 +/- 0.008860558923333883
lpips 0.040237267 +/- 0.0024190666200593113

Do I miss something?

Thanks.

**Comment by White-Link** [comment]
Hi!

We ran our test code with the command given in the README and the pretrained models that we provide, and obtained the same results as in our article.

Without additional information, it is difficult for us to determine the problem that you encounter. What is the command that you executed to obtain the results that you report ? You might also consider checking that you use the same computing environment as ours (given in `requirements.txt`).

**Comment by kaanakan** [comment]
Hi,

thank you for the answer. 

The environment I am using has the same modules with the requirements file and it python 3.7.6 with cuda 10.1. The comment I use:

`
python test.py 
 --data_dir /datasets/BAIR 
 --xp_dir models 
 --nt_gen 30 
 --lpips_dir lpips_weight/ 
--device 0 
--batch_size 32 
`
The xp_dir consists the pre-trained weight and the config file you shared.

**Comment by White-Link** [comment]
We executed our test script with this command by cloning this repository, preprocessing the dataset and downloading the model from scratch, but we could not reproduce your issue as we still obtained the results reported in the paper. Your LPIPS score is particularly surprising as it is significantly better than the ones of all compared models in our work!

A possible explanation could still lie in your computing environment. Is it standard? We are operating on a Linux system and Nvidia GPUs such as Titan Xp, Titan V and RTX 2080 Ti.

In order to help us investigate your issue, could you please provide examples of ground truth and model predictions so that we can examine them? These are saved as NumPy arrays by the test script in the model folder.

**Comment by kaanakan** [comment]
Hi,

I am using Linux environment with Tesla V100 (32 gb) gpu. I will upload conda environment. I also upload the results and prediction to drive, the link is below. I only uploaded ssim, psnr and lpips best predictions.

[env_info.txt](https://github.com/edouardelasalles/srvp/files/5622373/env_info.txt)
[results.zip](https://drive.google.com/file/d/1I72Qv3EyloOPgm9xB3POl0uqlkzXBXCO/view?usp=sharing)

Thanks.

**Comment by White-Link** [comment]
We noticed a mismatch between your predicted sequences and the ground truth on our side. This might explain the issue. Did you follow the steps described [here](https://github.com/edouardelasalles/srvp#bair) to preprocess the dataset?

If you want us to further investigate this matter, you can send us your BAIR testing data.

**Comment by kaanakan** [comment]
Thank you for the feedback. I think our dataset was processed by the code from svg repository and I thought it was going to same with your dataset since in paper, it is reported that you are using the same train and test set. I will probably process the dataset again and report the resullts on that.

Thank a lot.

**Comment by White-Link** [comment]
No worries!

We follow the exact same procedure as SVG to preprocess the dataset. Therefore, its test set should in principle be the same than ours; it is indeed the case on our side. Furthermore, the results that we report for all models being consistent with those reported, for instance, in [the SAVP article](https://arxiv.org/abs/1804.01523) indicates that we also use the same dataset as them.

Please let us know the results that you obtain; we are interested in resolving any technical issue in the reproduction of results of all models in our experimental evaluation.

**Comment by kaanakan** [comment]
Hi again,

I preprocessed the data files that were downloaded before and got the results below. The PSNR and SSIM results are similar but again the LPIPS score differs too much. Here are the results:

Results:
psnr 19.56701 +/- 0.26613611459732056
ssim 0.82046306 +/- 0.008313765414059163
lpips 0.035872065 +/- 0.0020749701419845224

I will download the dataset and preprocess it again. Then, I will evaluate the model.

**Comment by kaanakan** [comment]
I downloaded the dataset and preprocess it with using your repository. The results are exactly same. 

Results:
psnr 19.56701 +/- 0.2661360853910446
ssim 0.82046306 +/- 0.008313765414059163
lpips 0.035872065 +/- 0.0020749701419845224

My environment again matches with the requirements file. The run command is below:

`
python test.py 
 --data_dir /datasets/BAIR/srvp_version 
 --xp_dir models 
 --nt_gen 30 
 --lpips_dir lpips_weight/ 
--device 0 
--batch_size 32 
`
 
I also upload the predictions, ssim, psnr, lpips best predictions.
[results](https://drive.google.com/file/d/1OMYCT4dm3rZQuul-qbETeK3dAH00yOFH/view?usp=sharing)

If you have any other suggestions,  I will try them.
Thanks a lot.

**Comment by White-Link** [comment]
Thank you for the update!

Regarding the issue with respect to LPIPS, we think that you use the outdated LPIPS weigthts. Indeed, we used the most recent ones ([v0.1](https://github.com/richzhang/PerceptualSimilarity/tree/master/lpips/weights/v0.1)). When we tested our model with the older ones ([v0.0](https://github.com/richzhang/PerceptualSimilarity/tree/master/lpips/weights/v0.0)), we actually obtained the same results that you report.

We are sorry about the confusion: we specified the LPIPS version in our code only. We just modified the README.md file to clarify this point.

Can you confirm that using the most recent LPIPS weights makes your results match ours?

**Comment by kaanakan** [comment]
You are correct.

I guessed the version considering the publication date and I guessed it wrong :)

The results I got with v0.1:

psnr 19.56701 +/- 0.26613611459732056
ssim 0.8204631 +/- 0.008313765414059163
lpips 0.057495013 +/- 0.0031978254998102783

Thank you for your time and effort.

**Comment by White-Link** [comment]
You're welcome! We're now closing the issue as the problem was solved. Feel free to contact us if you have any other question.
